crates/ruff_linter/src/rules/flake8_type_checking/rules/type_alias_quotes.rs

```diff
@@ -0,0 +1,280 @@
+use crate::registry::Rule;
+use ast::{ExprContext, Operator};
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast as ast;
+use ruff_python_ast::{Expr, Stmt};
+use ruff_python_semantic::{Binding, SemanticModel};
+use ruff_text_size::{Ranged, TextRange};
+use std::borrow::Borrow;
+
+use crate::checkers::ast::Checker;
+use crate::rules::flake8_type_checking::helpers::quote_type_expression;
+
+/// ## What it does
+/// Checks if [PEP 613] explicit type aliases contain references to
+/// symbols that are not available at runtime.
+///
+/// ## Why is this bad?
+/// We will get a `NameError` at runtime.
+///
+/// ## Example
+/// ```python
+/// from typing import TYPE_CHECKING, TypeAlias
+///
+/// if TYPE_CHECKING:
+///     from foo import Foo
+/// OptFoo: TypeAlias = Foo | None
+/// ```
+///
+/// Use instead:
+/// ```python
+/// from typing import TYPE_CHECKING, TypeAlias
+///
+/// if TYPE_CHECKING:
+///     from foo import Foo
+/// OptFoo: TypeAlias = "Foo | None"
+/// ```
+///
```

Comment:
Why is the fix marked as unsafe? We can document this in a `## Fix safety` section,

Reply:
Good question, I went by the precedent set by `runtime_import_in_type_checking_block` which uses the same function to quote annotations in its fix and also doesn't explain its reasoning. So I don't know what the original reasoning was. It might just be the fact that the forward reference would not be able to be resolved at runtime. Which is a problem with flake8-type-checking in general.

Reply:
Looks like it is at least in part because it can remove comments on multi-line expressions. Maybe this documentation gap should first be addressed in a separate PR though, since it also affects other existing rules.